### PR TITLE
CVSL-741 Tidying up endpoints

### PR DESCRIPTION
**Pull request by andrewrlee**
1.) Removing added/removed values from the the policy version compare endpoints

2.) Adding shortCategory to PSS additional conditions

3.) Removing the unused policy compare endpoint

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/policy/AdditionalConditionPss.kt [hunk 1]
```diff
@@ -6,6 +6,7 @@ data class AdditionalConditionPss(
   override val text: String,
   override val tpl: String?,
   override val requiresInput: Boolean,
+  override val categoryShort: String?,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/policy/AdditionalConditionPss.kt:9]: To help @SebNegahbanMoJ work with conditions using polymorphism.

( sounds like there's no good reason to not include this from a modelling point of view)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicencePolicyController.kt [hunk 2]
```diff
@@ -133,21 +132,6 @@ class LicencePolicyController(
...
-  @GetMapping(value = ["/compare/{version1}/{version2}"])
-  @PreAuthorize("hasAnyRole('SYSTEM_USER', 'CVL_ADMIN')")
-  @ResponseBody
-  @Operation(
-    summary = "Get licence policy differences between two versions",
-    description = "Returns a list of added, remove or amended policy conditions between two versions" +
-      "Requires ROLE_SYSTEM_USER or ROLE_CVL_ADMIN.",
-    security = [SecurityRequirement(name = "ROLE_SYSTEM_USER"), SecurityRequirement(name = "ROLE_CVL_ADMIN")],
-  )
-  fun compare(@PathVariable("version1") version1: String, @PathVariable("version2") version2: String): PolicyChanges? {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicencePolicyController.kt:145]: unused endpoint

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicencePolicyService.kt [hunk 3]
```diff
@@ -46,8 +35,8 @@ data class LicenceConditionChanges(
   val sequence: Int?,
   val previousText: String,
   val currentText: String?,
-  var addedInputs: List<Any>,
-  var removedInputs: List<Any>,
+  @JsonIgnore var addedInputs: List<Any>,
+  @JsonIgnore var removedInputs: List<Any>,
   val suggestions: List<SuggestedCondition> = emptyList()
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicencePolicyService.kt:40]: including these for the `PolicyVersionDifferencesTest` but excluding from the API as they aren't currently being used and causing inconveniences on the frontend.

**Review by mgreen-mojhub — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/PolicyFunctions.kt [hunk 4]
```diff
@@ -10,28 +10,6 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.ConditionCh
...
-  current.filter { pssElement ->
-    !other.any { pss -> pss.code == pssElement.code }
-  }
-
-fun <T : ILicenceCondition> addedConditions(current: List<T>, other: List<T>): List<T> =
-  other.filter { pssElement ->
-    !current.any { pss -> pss.code == pssElement.code }
-  }
-
-fun <T : ILicenceCondition> amendedConditions(
```
Comment (SebNegahbanMoJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/PolicyFunctions.kt:23]: I assume these aren't shared methods with the licence change endpoint that we do use as a part of the inflight stuff?

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/PolicyFunctions.kt:23]: no, we'd get compilation failures if they were used elsewhere

**Pull request merged**